### ISSUE TITLE
Log Curve Info for 1411 does not report class index correctly

### ISSUE THREAD

**Issue by TessForGithub2** [hashmapinc/Drillflow#460]
Associated card is #454 
Currently the log curve info just puts in 0 for class index, when it should reflect the index of the value in the csv portion of logdata.
Clarification:
v1.4.1.1 has classIndex (this is not the case for v1.3.1.1; v1.3.1.1 has columnIndex -- there is be a separate issue for this version)
classIndex is present in the POJO & defaults to 0; need to just place the correct value into this field

**Comment by Mike-d-s** [comment]
Theresa, please revise this card and break it into two separate cards, one for <classIndex/> for 1.4.1.1 which is part of the log curve metadata and one for <columnIndex/> for 1.3.1.1 which is the server side attribute specifying the order of the log curve in the data table by column count.

**Comment by TessForGithub2** [comment]
Update: I put in the fix and tested it (it looks fine). However, I want to perform the integration test, but "get channels" is reporting only classIndex (and not columnIndex). I will open a card for "get channels".

**Comment by Mike-d-s** [comment]
EPIC:
#361 Log CRUD Implementation

**Comment by TessForGithub2** [comment]
Update: I did as Mike suggested & did a Postman test using DoT's version of Get Channels. I have an interesting situation where DoT is returning "classIndex" instead of "columnIndex" although, when I trace what I sent over, it is 1.3.1.1 and columnIndex. I will open a new card on this, since our Get Channels may not be the issue (I will alert Sukhe next). I also will perform another trace (but I did show Shehzad that I am sending in the payload for Channels what I think I should be sending...he was an extra pair of eyes to verify I was doing it correctly).

**Comment by TessForGithub2** [comment]
Update: DoT is fine, it was my understanding that was faulty. DoT only speaks "1.4.1.1". Drillflow performs all required conversions to speak "1.3.1.1" or "2.0". Cards updated to reflect this correction in thought.

**Comment by Mike-d-s** [comment]
Blocked waiting on fix from Sukhe.  Additionally, needs to get valid test case from Shehzad.

**Comment by niaalex** [comment]
The fix is in and the team will address today in the merge.  ETA 6/17.

**Comment by TessForGithub2** [comment]
UPDATE:

In Channel.java, 1 line of code needs to replace where classIndex is set to 0:
lci.setClassIndex(c.getClassIndex());

Will put this fix in today (June 18).

**Comment by TessForGithub2** [comment]
Update: merged this fix into the master branch; due to release 7/11.

**Comment by TessForGithub2** [comment]
Update: for 1.4.1.1, <columnIndex/> is part of the log curve metadata
Translation: 
-> from1411 (which gets WITSML XML from a v1.4.1.1 Client) just requires that <classIndex/> is populated from log curve metadata (the "lci") - already in Master
-> to1411 (which gets JSON from DoT) just requires that <classIndex/> is populated from returned channel's <classIndex/> - already in Master

Reopened this card to add documentation within the code base to be clear, and also clear up the language of this card.

**Comment by niaalex** [comment]
Covered in the 395A Branch fix.  ETA 7/22

**Comment by shehzadsidi** [comment]
Issue still exists, ClassIndex values are all 0

**Comment by TessForGithub2** [comment]
This is one I have already solved and is indf-395A. Will move it in quickly.

Theresa
